File: ClientApp/src/app/posts/upvote.ts
import { IPost, Post } from "./post";

export interface IUpvote {
  UpvoteId: number;
  Vote: string;
  PostId: number;
  UserId: number;
  post: IPost;
}
export class Upvote implements IUpvote {
    UpvoteId: number = 0;
    Vote: string = "";
    PostId: number = 0;
    UserId: number = 0;
    post: IPost = new Post();
}

File: WebAppAngular/Models/Upvote.cs
using Newtonsoft.Json;

namespace OsloMetAngular.Models;

/* DATA MODEL (SCHEMA) FOR VOTES */
// User and Post are navigation properties.
public class Upvote
{
    // VoteID
    [JsonProperty(nameof(UpvoteId))]
    public int UpvoteId { get; set; }

    // The actual vote data (up or down)
    [JsonProperty(nameof(Vote))]
    public string Vote { get; set; } = string.Empty;

    // UserID
    [JsonProperty(nameof(UserId))]
    public int UserId { get; set; }

    // Virtual user
    public virtual User User { get; set; } = default!;

    // PostID
    [JsonProperty(nameof(PostID))]
    public int PostID { get; set; }

    // Virtual post
    public virtual Post Post { get; set; } = default!;     
}
